### PR TITLE
Adds Dockerfile and K8s resources for gitops-backend service and RDBMS (Solves: Gitops-1380)

### PR DESCRIPTION
**Pull request by samyak-jn**
Hello! :hugs:

This is a PR in regards to solving issue [GITOPS-1380](https://issues.redhat.com/browse/GITOPS-1380). The PR consists of following changes:
- Adds a Dockerfile
- 'make image' and 'make image-push' target to the Makefile
- Adds K8s resources (Deployment, Service, Route) 
- Adds K8s resources for development postgresql

### CODE REVIEW

File: backend/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+# Copy the go source
+COPY *.go ./
+COPY config/ config/
+COPY routes/ routes/
+COPY util/ util/
+
+# To compile/build the go applications
+RUN make build
+RUN ./dist/managed-gitops-backend
```
Comment (sbose78) [inline backend/Dockerfile:21]: Wondering why we are executing the binary here?
Could we do a multi-stage Dockerfile build where..
* In stage 1, we build the binary and,
* In stage 2, we pack that binary into a lean public image ( like registry.access.redhat.com/ubi8/ubi-minimal ) for deployment.

File: backend/Makefile [hunk 2]
```diff
@@ -1,3 +1,5 @@
+IMAGE_NAME?=gitops-backend
```
Comment (jgwest) [inline backend/Makefile:1]: I've created a new repo for us to use:
https://quay.io/repository/redhat-appstudio/gitops-service?tab=tags

So IMAGE_NAME can now be quay.io/redhat-appstudio/gitops-service

File: manifests/services/application/deplyoment.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    matchLabels:
+      app.kubernetes.io/name: managed-gitops
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: managed-gitops
+    spec:
+      containers:
+      - name: managed-gitops
+        image: quay.io/samyakjain/managed-gitops:latest
```
Comment (jgwest) [inline manifests/services/application/deplyoment.yaml:16]: I setup a new repo for us to use, and sent you an invite: https://quay.io/repository/redhat-appstudio/gitops-service?tab=tags

See if you can push to there, then this line can be updated to `quay.io/redhat-appstudio/gitops-service:latest`

**Review by jgwest — CHANGES_REQUESTED** [review]
LGTM, some minor tweaks then good to go.

File: manifests/services/application/deplyoment.yaml (hunk 3 above)
Comment (samyak-jn) [reply manifests/services/application/deplyoment.yaml:16]: Got the invite, working, thanks! :dango:

**Pull request merged**